ReferenceQuality.pm

```diff
@@ -0,0 +1,194 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+=head1 CONTACT
+
+ Ensembl <http://www.ensembl.org/info/about/contact/index.html>
+
+=cut
+
+=head1 NAME
+
+ ReferenceQuality
+
+=head1 SYNOPSIS
+
+ mv ReferenceQuality.pm ~/.vep/Plugins
+ ./vep -i variations.vcf --plugin ReferenceQuality,/path/to/data.gff3.gz
+
+=head1 DESCRIPTION
+
+ This is a plugin for the Ensembl Variant Effect Predictor (VEP) that
+ assesses the quality of the reference genome at the location of your variants.
+
+ The following steps are necessary before running this plugin:
+ 
+ GRCh38:
+ 
+ Download
+ ftp://ftp.ncbi.nlm.nih.gov/pub/grc/human/GRC/GRCh38/MISC/annotated_clone_assembly_problems_GCF_000001405.38.gff3
+ ftp://ftp.ncbi.nlm.nih.gov/pub/grc/human/GRC/Issue_Mapping/GRCh38.p9_issues.gff3
+ 
+ cat annotated_clone_assembly_problems_GCF_000001405.38.gff3 GRCh38.p9_issues.gff3 > GRCh38_quality_mergedfile.gff3
+ sort -k 1,1 -k 4,4n -k 5,5n GRCh38_quality_mergedfile.gff3 > sorted_GRCh38_quality_mergedfile.gff3
+ bgzip sorted_GRCh38_quality_mergedfile.gff3
+ tabix -p gff sorted_GRCh38_quality_mergedfile.gff3.gz
+
+ The plugin can then be run with:
+ ./vep -i variations.vcf --plugin ReferenceQuality,sorted_GRCh38_quality_mergedfile.gff3.gz
+  
+ GRCh37:
+ 
+ Download
+ ftp://ftp.ncbi.nlm.nih.gov/pub/grc/human/GRC/GRCh37/MISC/annotated_clone_assembly_problems_GCF_000001405.24.gff3
+ ftp://ftp.ncbi.nlm.nih.gov/pub/grc/human/GRC/Issue_Mapping/GRCh37.p9_issues.gff3
+ 
+ cat annotated_clone_assembly_problems_GCF_000001405.24.gff3 GRCh37.p9_issues.gff3 > GRCh37_quality_mergedfile.gff3
+ sort -k 1,1 -k 4,4n -k 5,5n GRCh37_quality_mergedfile.gff3 > sorted_GRCh37_quality_mergedfile.gff3
+ bgzip sorted_GRCh37_quality_mergedfile.gff3
+ tabix -p gff sorted_GRCh37_quality_mergedfile.gff3.gz
+ 
+ The plugin can then be run with:
+ ./vep -i variations.vcf --plugin ReferenceQuality,sorted_GRCh37_quality_mergedfile.gff3.gz
+ 
+ 
+
+ The tabix utility must be installed in your path to use this plugin.
+
+=cut
+package ReferenceQuality;
+
+use strict;
+use warnings;
+
+use Bio::EnsEMBL::Utils::Sequence qw(reverse_comp);
+
+use Bio::EnsEMBL::Variation::Utils::BaseVepTabixPlugin;
+
+use base qw(Bio::EnsEMBL::Variation::Utils::BaseVepTabixPlugin);
+
+sub new {
+  my $class = shift;
+
+  my $self = $class->SUPER::new(@_);
+
+  $self->expand_left(0);
+  $self->expand_right(0);
+
+  $self->get_user_params();
+
+  return $self;
+}
+
+sub feature_types {
+  return ['Transcript'];
+}
+
+sub get_header_info {
+  return { ReferenceQuality => 'Indicates quality of reference genome at input position'};
+}
+
+sub run {
+  my ($self, $tva) = @_;
+    
+  my $vf = $tva->variation_feature;
+  my $allele = $tva->variation_feature_seq;
+  my $chr = $vf->{chr};
+
+  my $chr_syn;
+  my @new_chr_array;
+  
+  #The NC_ chromosome synonym is found differently is we have database access
```

Comment:
It would be quicker to cache the synonyms rather than look them up from file or db for each line.